trieste/models/gpflux/models.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import warnings
+from typing import Any, Dict
+
+import tensorflow as tf
+from gpflux.layers import GPLayer
+from gpflux.models import DeepGP
+from gpflux.models.deep_gp import sample_dgp
+
+from ...data import Dataset
+from ...types import TensorType
+from ...utils import jit
+from ..interfaces import TrainableProbabilisticModel
+from ..optimizer import Optimizer
+from .interface import GPfluxPredictor
+
+
+class DeepGaussianProcess(GPfluxPredictor, TrainableProbabilisticModel):
+    """
+    A :class:`TrainableProbabilisticModel` wrapper for a GPflux :class:`~gpflow.models.DeepGP` with
+    only standard :class:`GPLayer`: this class does not support keras layers, latent variable
+    layers, etc.
+
+    Note: the user should remember to set `tf.keras.backend.set_floatx()` with the desired value
+    (consistent with GPflow) so that dtype errors do not occur.
+    """
+
+    def __init__(
+        self,
+        model: DeepGP,
+        optimizer: Optimizer | None = None,
+        fit_args: Dict[Any] | None = None,
+        compile: bool | None = None,
+    ):
+        """
+        :param model: The underlying GPflux deep Gaussian process model.
+        :param optimizer: The optimizer with which to train the model. Defaults to
+            :class:`~trieste.models.optimizer.TFOptimizer` with :class:`~tf.optimizers.Adam`. Only
+            the optimizer itself is used; other args relevant for fitting should be passed as part
+            of `fit_args`.
+        :param fit_args: A dictionary of arguments to be used in the Keras `fit` method. Default to
+            using 100 epochs, batch size 100, and verbose 0.
+        :param compile: Whether to compile into a graph for sampling.
+        """
+
+        super().__init__()
+
+        if isinstance(optimizer, Optimizer):
```

Comment:
why do you kepp the Optimizer argument?

Reply:
I got confused and thought the typing wouldn't allow it, but it actually is fine. It's done now